R/look_up.R

```diff
@@ -0,0 +1,371 @@
+#' @param assign_with_GTS \code{character} or \code{FALSE}. Allows intervals to
+#' be searched in the `GTS2020` (default) or the `GTS2012` table. Set to
+#' \code{FALSE} to disable.
+#' @param early_interval \code{character}. Name of the column in `occdf` that
+#' contains the earliest interval from which the occurrences are from. Defaults
+#'  to `early_interval`.
+#' @param late_interval \code{character}. Name of the column in `occdf` that
+#' contains the latest interval from which the occurrences are from. Defaults
+#'  to `late_interval`.
+#' @param print_assigned \code{logical}. Should the assigned interval names be
+#' printed? Defaults to \code{FALSE}.
+#'
+#' @return A \code{dataframe} of the original input `data` with the following
+#' appended columns is returned: `early_stage` and `late_stage`, corresponding
+#' to the earliest and latest international geological stage which
+#' could be assigned to the occurrence based on the given interval names.
+#' `interval_max_ma` and `interval_min_ma` return maximal and minimal interval
+#' ages if provided in the interval key, or if they can be fetched from GTS2012
+#' or GTS2020. A column `interval_mid_ma` is appended to provide the midpoint
+#' age of the interval.
+#'
+#' @details
+#' Instead of  geological stages, the user can supply any names in the
+#' `early_stage` and `late_stage` column of `int_key`.
+#' `assign_with_GTS` should then be set to \code{FALSE}.
+#'
+#' An exemplary `int_key` has been included within this package
+#' (\code{\link{interval_key}}). This key works well for assigning
+#' geological stages to many of the intervals from the
+#' [Paleobiology Database](https://paleobiodb.org)
+#' and the [Paleoreefs Database](https://www.paleo-reefs.pal.uni-erlangen.de/).
+#' Palaeoverse can provide no guaranty that all of
+#' the stage assignments with the exemplary key are accurate.
+#' The table corresponding to this key can be loaded with
+#' `palaeoverse::interval_key`.
+#'
+#'
+#' @section Developer(s):
+#' Kilian Eichenseer & William Gearty
+#' @section Reviewer(s):
+#' Lewis A. Jones
+#' @examples
+#' # Grab internal tetrapod data
+#' occdf <- tetrapods
+#' # assign stages using the exemplary interval_key
+#' occdf <- look_up(occdf)
+#'
+#' ## Use own key to assign intervals
+#' # create example data
+#' occdf <- data.frame(
+#'            stage = c("any Permian", "first Permian stage",
+#'                      "any Permian", "Roadian"))
+#' # create example key
+#' interval_key <- data.frame(
+#'                   interval_name = c("any Permian", "first Permian stage"),
+#'                   early_stage = c("Asselian", "Asselian"),
+#'                   late_stage = c("Changhsingian", "Asselian"))
+#' # assign stages using the custom interval_key, use "GTS2012":
+#' occdf <- look_up(occdf, int_key = interval_key, assign_with_GTS = "GTS2012",
+#' early_interval = "stage", late_interval = "stage",
+#' print_assigned = TRUE)
+#' @export
```

Comment:
I think it would also be sensible to include an example without the interval key - I think a lot of people are going to use this function to transform character based interval names into numeric ones, and won't necessarily be using their own int_key to do so. I know it's obvious, but I think it would be best for completion's sake.